packages/hardhat/test/ChainLance.ts

```diff
@@ -13,15 +13,330 @@ describe("ChainLance", function () {
+      .connect(employer2)
+      .createProject(canceledProjectId, canceledProjectPrice, 100);
+    await txCreateCanceledProject.wait();
+
+    const bidId5 = "bidid5";
+    await chainLance.connect(worker1).bidProject(canceledProjectId, bidId5, canceledProjectPrice, 1);
+
+    const bidId6 = "bidid6";
+    await chainLance.connect(worker2).bidProject(canceledProjectId, bidId6, canceledProjectPrice, 1);
+
+    const bidId7 = "bidid7";
+    await chainLance.connect(worker3).bidProject(canceledProjectId, bidId7, canceledProjectPrice, 1);
+
+    await chainLance.connect(employer2).cancelProject(canceledProjectId);
+  });
+
+  it("Should list projects in different states", async function () {
+    const openProjects = await chainLance.listProjectsWithState(0);
+    expect(openProjects).to.include("id42");
+    expect(openProjects).is.lengthOf(1);
+
+    const inWorkProjects = await chainLance.listProjectsWithState(1);
+    expect(inWorkProjects).to.include("id3");
+    expect(inWorkProjects).is.lengthOf(1);
+
+    const inReviewProjects = await chainLance.listProjectsWithState(2);
+    expect(inReviewProjects).to.include("id4");
+    expect(inReviewProjects).is.lengthOf(1);
+
+    const completedProjects = await chainLance.listProjectsWithState(3);
+    expect(completedProjects).to.include("id5");
+    expect(completedProjects).is.lengthOf(1);
+
+    const canceledProjects = await chainLance.listProjectsWithState(4);
+    expect(canceledProjects).to.include("id6");
+    expect(canceledProjects).is.lengthOf(1);
+  });
+});
+
+describe("ChainLance: check access rights", function () {
+  let chainLance: ChainLance;
+  before(async () => {
+    const chainLanceFactory = await ethers.getContractFactory("ChainLance");
+    chainLance = (await chainLanceFactory.deploy()) as ChainLance;
+    await chainLance.waitForDeployment();
+
+    const [employer1, employer2, employer3, worker1, worker2, worker3] = await ethers.getSigners();
+    ///////////////////////////////////////////////////////////////////
+    // Work with state: Open
+    const noBidsProjectId = "id42";
+    const noBidsProjectPrice = 1000;
+    const txCreateNoBidsProject = await chainLance
+      .connect(employer1)
+      .createProject(noBidsProjectId, noBidsProjectPrice, 100);
+    await txCreateNoBidsProject.wait();
+    ///////////////////////////////////////////////////////////////////
+    // Work with state: InWork
+    const oneBidProjectId = "id3";
+    const oneBidProjectPrice = 350000;
+    const txCreateOneBidProject = await chainLance
+      .connect(employer1)
+      .createProject(oneBidProjectId, oneBidProjectPrice, 100);
+    await txCreateOneBidProject.wait();
+
+    const bidId1 = "bidid1";
+    await chainLance.connect(worker1).bidProject(oneBidProjectId, bidId1, oneBidProjectPrice, 1);
+
+    const txAcceptBid1 = await chainLance
+      .connect(employer2)
+      .acceptBid(oneBidProjectId, bidId1, { value: oneBidProjectPrice });
+    await txAcceptBid1.wait();
```

Comment:
We  should  check contract call is failed here. Get receipt for this transaction (`const receipt = await txAcceptBid1.wait();`) and check receipt.state is 2 (`expect(receipt.state).to.be equal(2, "contract call should fail")`)